python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -150,24 +99,66 @@ def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
     def add_kwargs_from_args(
         cls, args: argparse.Namespace, kwargs: typing.Dict[str, typing.Any]
     ) -> None:
-        """Process custom --hostname and --unit arguments."""
+        """Process custom --hostname, --port and --unit arguments."""
         cls.hostname = (
             f"m1m3cam-aircomp{kwargs['index']:02d}.cp.lsst.org"
             if args.hostname is None
             else args.hostname
         )
+        cls.port = args.port
         cls.unit = kwargs["index"] if args.unit is None else args.unit
 
     async def close_tasks(self) -> None:
+        await super().close_tasks()
         if self.simulation_mode == 1:
             await self.simulator.shutdown()
             await self.simulator_future.cancel()
-        await super().close_tasks()
-
-    async def begin_start(self, data):
+        self.poll_task.cancel()
```

Comment:
When you cancel simulator_future above you don't check if it's None first. If you change the initializer to make it a done future then that's fine. But if you still want to initialize it to None then you have to check it against None *everywhere* you act on it.

Reply:
I check if I run in simulator. If I do, simulator_future will be intialized.